### PR TITLE
SCALRCORE-24801 Integrations > Webhook update provider resource

### PR DESCRIPTION
**Pull request by petroprotsakh**
### Changelog
* [x] I have updated CHANGELOG.md according to [CONTRIBUTING.md](../CONTRIBUTING.md)

### Documentation
* [x] I have updated resource/data source documentation in [docs](../docs)

### State
* [x] My changes affect the state
* [x] I have included a state migration and a unit test for it

### CODE REVIEW

**Review by DayS1eeper — CHANGES_REQUESTED** [review]

File: scalr/resource_scalr_webhook.go [hunk 1]
```diff
@@ -213,25 +437,51 @@ func resourceScalrWebhookRead(ctx context.Context, d *schema.ResourceData, meta
...
+
+	if webhook.IsShared {
+		_ = d.Set("environments", []string{"*"})
+	} else {
+		environmentIDs := make([]string, 0)
+		for _, environment := range webhook.Environments {
+			environmentIDs = append(environmentIDs, environment.ID)
+		}
+		_ = d.Set("environments", environmentIDs)
 	}
```
[the remark this answers is not in the document]
Comment (petroprotsakh) [reply scalr/resource_scalr_webhook.go:451]: @DayS1eeper It sets the environments to `[]`, so yes.

File: scalr/data_source_scalr_webhook.go [hunk 2]
```diff
@@ -82,55 +130,93 @@ func dataSourceScalrWebhookRead(ctx context.Context, d *schema.ResourceData, met
...
 		}
 	}
+	// Having the webhook found, read from old API then
+	// to populate deprecated fields available only in old API
+	oldWebhook, err := scalrClient.Webhooks.Read(ctx, newWebhook.ID)
+	if err != nil {
+		if err != nil {
+			return diag.Errorf("Error retrieving webhook: %v", err)
+		}
+	}
```
Comment (DayS1eeper) [inline scalr/data_source_scalr_webhook.go:167]: typo

**Review by DayS1eeper — APPROVED** [review]

**Comment by emocharnik** [comment]
@petroprotsakh, Pull request has conflicts with base branch

**Comment by emocharnik** [comment]
@petroprotsakh, Pull request has conflicts with base branch

**Comment by emocharnik** [comment]
@petroprotsakh, Some non-required builds are failing or stuck.

**Pull request merged**